java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -68,7 +75,14 @@ protected ColumnView(ColumnVector.OffHeapState state) {
     type = DType.fromNative(ColumnView.getNativeTypeId(viewHandle), ColumnView.getNativeTypeScale(viewHandle));
     rows = ColumnView.getNativeRowCount(viewHandle);
     nullCount = ColumnView.getNativeNullCount(viewHandle);
-    AssertEmptyNulls.assertNullsAreEmpty(this);
+    try {
+      AssertEmptyNulls.assertNullsAreEmpty(this);
+    } catch (AssertionError ae) {
+      // This constructor is only called from ColumnVector so calling close will call
+      // close in ColumnVector which will close out the offHeapState
```

Comment:
That means calling derived class method in base class constructor. I suspect that it may be incorrect or won't produce the desired effect. What I'm seeing is only the `close` method of this class is called, then the exception is propagated to the derived class without calling `close` of the derived class.

Reply:
> That means calling derived class method in base class constructor.

No, not always. Only if the object is of the `ColumnVector` type, if the object is of ColumnView type then the `close()` in `ColumnView` will be called.
> What I'm seeing is only the close method of this class is called, then the exception is propagated to the derived class without calling close of the derived class

```
ColumnView c = new ColumnVector()
c.close();
```
^ this will call the `close()` in `ColumnVector`

```
ColumnView c = new ColumnView()
c.close();
```
^ this will call `close()` in `ColumnView`
In both cases we will get the desired results

Reply:
Yeah, but the example above is showing that `close()` is called on an already-constructed object. What I mentioned above is `close()` called during object construction. When an instance of the derived class is being constructed, the constructor of the base class is called first (`super(....)`) in the constructor of the derived class. Since the construction of the derived class is not yet completed, polymorphism doesn't take place.

Reply:
It seems that the same rule should be followed in Java: https://stackoverflow.com/questions/3404301/whats-wrong-with-overridable-method-calls-in-constructors

Reply:
+1 we should not rely on close. In this case, `ColumnView.close` is also a noop for a straight `ColumnView` at this time, as it checks that `offHeap` is null, and only then it takes action.

Reply:
@ttnghia @abellina thank you for taking a look at this. 

I lost sight in the second case that it is a constructor and will not behave properly. Will fix